packages/teams-js/src/private/files.ts

```diff
@@ -529,4 +593,206 @@ export namespace files {
 
     sendMessageToParent('files.openDownloadFolder', [fileObjectId], callback);
   }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates add 3P cloud storage provider flow
+   *
+   * @param callback Callback that will be triggered post add 3P cloud storage provider action
+   */
+  export function add3PCloudStorageProvider(callback: (error?: SdkError) => void): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!callback) {
+      throw new Error('[files.add3PCloudStorageProvider] callback cannot be null');
+    }
+
+    sendMessageToParent('files.add3PCloudStorageProvider', [], callback);
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates signout of 3P cloud storage provider flow
+   *
+   * @param callback Callback that will be triggered post signout of 3P cloud storage provider action
+   */
+  export function remove3PCloudStorageProvider(
+    logoutRequest: I3PCloudStorageProviderRequest<I3PCloudStorageProviderLogoutRequestContentType>,
+    callback: (error?: SdkError) => void,
+  ): void {
+    ensureInitialized(FrameContexts.content);
+
+    if (!validateNonEmptyContent(logoutRequest, callback)) {
+      return;
+    }
+    if (!validateArgumentConsistency(logoutRequest, callback)) {
+      return;
+    }
+    if (!callback) {
+      throw new Error('[files.remove3PCloudStorageProvider] callback cannot be null');
+    }
+
+    sendMessageToParent('files.remove3PCloudStorageProvider', [logoutRequest], callback);
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   *
+   * Initiates the 3P cloud storage provider file action (New / Rename / Upload / Download / Delete)
```

Comment:
More detail in this comment about what these actions actually do or should be used for would be very helpful